bundles/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/internal/handler/LGWebOSHandler.java

```diff
@@ -147,13 +161,16 @@ public void dispose() {
         stopKeepAliveJob();
         stopReconnectJob();
 
+        discoveryServiceRegistry.removeDiscoveryListener(this);
+
         LGWebOSTVSocket s = socket;
         if (s != null) {
             s.setListener(null);
-            scheduler.execute(() -> s.disconnect()); // dispose should be none-blocking
+            s.disconnect();
         }
         socket = null;
         config = null; // ensure config gets actually refreshed during re-initialization
+        disposed = true;
```

Comment:
why is this necessary?

Reply:
This is the code from the PR #7299 
I integrated the other PR in this one for practical reason (for testing with all my changes) but this will be removed when this PR will be ready for a merge.
So please consider moving your comment in the right PR.

This is necessary because the websocket thread is closed asynchronously and so can call the thing handler (ConfigProvider) while the thing handler is already disposed.

Reply:
Removed from this PR